src/rust/engine/process_execution/src/remote_cache_tests.rs

```diff
@@ -243,13 +259,58 @@ async fn cache_read_eager_fetch() {
   assert_eq!(eager_local_call_count, 1);
 }
 
+#[tokio::test]
+async fn cache_read_speculation() {
+  WorkunitStore::setup_for_tests();
+
+  async fn run_process(local_delay_ms: u64, remote_delay_ms: u64, cache_hit: bool) -> (i32, usize) {
```

Comment:
Rather than using `_ms` suffixed values integers, prefer to use `Duration`, which you can create like `Duration::from_secs`, `Duration::from_millis`, etc: https://doc.rust-lang.org/std/time/struct.Duration.html

Reply:
I tried doing this but I think it made our test code more verbose and worse:

```diff
diff --git a/src/rust/engine/process_execution/src/remote_cache_tests.rs b/src/rust/engine/process_execution/src/remote_cache_tests.rs
index 28f8b055e..c1ee88110 100644
--- a/src/rust/engine/process_execution/src/remote_cache_tests.rs
+++ b/src/rust/engine/process_execution/src/remote_cache_tests.rs
@@ -36,7 +36,7 @@ impl MockLocalCommandRunner {
   pub fn new(
     exit_code: i32,
     call_counter: Arc<AtomicUsize>,
-    delay_ms: u64,
+    delay: Duration,
   ) -> MockLocalCommandRunner {
     MockLocalCommandRunner {
       result: Ok(FallibleProcessResultWithPlatform {
@@ -48,7 +48,7 @@ impl MockLocalCommandRunner {
         platform: Platform::current().unwrap(),
       }),
       call_counter,
-      delay: Duration::from_millis(delay_ms),
+      delay,
     }
   }
 }
@@ -113,7 +113,7 @@ fn create_local_runner(
   let local_runner = Box::new(MockLocalCommandRunner::new(
     exit_code,
     call_counter.clone(),
-    delay_ms,
+    Duration::from_millis(delay_ms),
   ));
   (local_runner, call_counter)
 }
@@ -124,7 +124,7 @@ fn create_cached_runner(
   read_delay_ms: u64,
   eager_fetch: bool,
 ) -> (Box<dyn CommandRunnerTrait>, StubActionCache) {
-  let action_cache = StubActionCache::new(read_delay_ms).unwrap();
+  let action_cache = StubActionCache::new(Duration::from_millis(read_delay_ms)).unwrap();
   let runner = Box::new(
     crate::remote_cache::CommandRunner::new(
       local.into(),
@@ -503,7 +503,7 @@ async fn make_action_result_basic() {
     .expect("Error saving directory");
 
   let mock_command_runner = Arc::new(MockCommandRunner);
-  let action_cache = StubActionCache::new(0).unwrap();
+  let action_cache = StubActionCache::new(Duration::from_millis(0)).unwrap();
   let runner = crate::remote_cache::CommandRunner::new(
     mock_command_runner.clone(),
     ProcessMetadata::default(),
diff --git a/src/rust/engine/process_execution/src/remote_tests.rs b/src/rust/engine/process_execution/src/remote_tests.rs
index 2f5840042..6989874b7 100644
--- a/src/rust/engine/process_execution/src/remote_tests.rs
+++ b/src/rust/engine/process_execution/src/remote_tests.rs
@@ -1745,7 +1745,7 @@ async fn remote_workunits_are_stored() {
     .file(&TestData::roland())
     .directory(&TestDirectory::containing_roland())
     .build();
-  let action_cache = mock::StubActionCache::new(0).unwrap();
+  let action_cache = mock::StubActionCache::new(std::time::Duration::from_millis(0)).unwrap();
   let (command_runner, _store) =
     create_command_runner(action_cache.address(), &cas, Platform::Linux);
 
@@ -2301,7 +2301,8 @@ async fn extract_execute_response(
   operation: Operation,
   remote_platform: Platform,
 ) -> Result<RemoteTestResult, ExecutionError> {
-  let action_cache = mock::StubActionCache::new(0).expect("failed to create action cache");
+  let action_cache =
+    mock::StubActionCache::new(Duration::from_millis(0)).expect("failed to create action cache");
 
   let cas = mock::StubCAS::builder()
     .file(&TestData::roland())
diff --git a/src/rust/engine/testutil/mock/src/action_cache.rs b/src/rust/engine/testutil/mock/src/action_cache.rs
index 234ce6327..1395acb61 100644
--- a/src/rust/engine/testutil/mock/src/action_cache.rs
+++ b/src/rust/engine/testutil/mock/src/action_cache.rs
@@ -135,13 +135,13 @@ impl ActionCache for ActionCacheResponder {
 }
 
 impl StubActionCache {
-  pub fn new(read_delay_ms: u64) -> Result<Self, String> {
+  pub fn new(read_delay: Duration) -> Result<Self, String> {
     let action_map = Arc::new(Mutex::new(HashMap::new()));
     let always_errors = Arc::new(AtomicBool::new(false));
     let responder = ActionCacheResponder {
       action_map: action_map.clone(),
       always_errors: always_errors.clone(),
-      read_delay: Duration::from_millis(read_delay_ms),
+      read_delay,
     };
 
     let addr = "127.0.0.1:0"
```

Were this production code, I would agree to keep it more flexible.